tidy3d/components/simulation.py

```
@@ -2657,6 +2657,71 @@ def _projection_monitors_distance(cls, val, values):
                         )
         return val
 
+    @pydantic.validator("monitors", always=True)
+    @skip_if_fields_missing(["size"])
+    def _projection_monitors_2d(cls, val, values):
+        """
+        Validate if the field projection monitor is set up for a 2D simulation and
+        ensure the observation angle is configured correctly.
+
+        - For a 2D simulation in the x-y plane, 'theta' should be set to 'pi/2'.
+        - For a 2D simulation in the y-z plane, 'phi' should be set to 'pi/2'.
+        - For a 2D simulation in the x-z plane, 'phi' should be set to '0'.
+
+        Note: Exact far field projection is not available yet. Currently, only
+        'far_field_approx = True' is supported.
+        """
+
+        if val is None:
+            return val
+
+        sim_size = values.get("size")
+
+        # Skip validation if 1D or 3D simulation
+        non_zero_dims = sum(1 for size in sim_size if size != 0)
+        if non_zero_dims == 1 or non_zero_dims == 3:
```

Does ProjectionMonitor work with 1D simulation? If not, is it validated somewhere already?

---

I haven't hear 1D simulation for far field projection, maybe we can just determine if it is for 2d like this?
    `non_zero_dims = sum(1 for size in sim.size if size != 0)
    if non_zero_dims == 2:`
          ...

---

i mean if it doesn't work with 1D simulation, we can throw an error.

---

Yeah, sure. I will raise an error if this isn't validated somewhere else.

---

Yeah, I don't think it's validated anywhere else.

Note also that ideally we should validate out the user-side projection if simulation is < 3D, until we fix that and remove the validation for 2D.